### PR TITLE
Provide a more useful error message to user if no CPLEX binary is found. 

### PR DESCRIPTION
**Pull request by dourouc05**
Instead of displaying a large message that is the same for all platforms, only show the relevant part for the user's platform. Moreover, check if some folder is recognised and propose it if it is. 

Maybe it would be good to indicate if the path that is proposed exists (and the proposed commands should work out of the box) or if the user should check further?

### CODE REVIEW

**Comment by dourouc05** [comment]
I must admit I don't understand the error message GitHub gives for the CI. Both commits come from the same fork!?

File: deps/build.jl [hunk 1]
```diff
@@ -16,6 +16,63 @@ function library_name(v)
...
+    """
+    
+    error_end = """
+    ```
+
+    See the CPLEX.jl README at https://github.com/jump-dev/CPLEX.jl for further
+    instructions.
+    """
+    
+    return error_beginning * error_middle * error_end
```
Comment (odow) [inline deps/build.jl:73]: Why split into three strings only to recombine them?

File: deps/build.jl [hunk 2]
```diff
@@ -16,6 +16,63 @@ function library_name(v)
...
+    cplex_studio_folders_mac = ["/Applications/$f/cplex/bin/x86-64_osx/" for f in cplex_studio_stem_folders]
+    cplex_studio_folders_lin = ["/opt/$f/cplex/bin/x86-64_linux/" for f in cplex_studio_stem_folders]
+    
+    cplex_studio_folders = if Sys.iswindows()
+        cplex_studio_folders_win 
+    elseif Sys.isapple()
+        cplex_studio_folders_mac
+    else
+        cplex_studio_folders_lin
+    end
```
Comment (odow) [inline deps/build.jl:32]: You could also do
```Julia
cplex_studio_folders = map(["CPLEX_Studio201", "CPLEX_Studio1210"]) do f
    if Sys.iswindows()
        return "C:\\\\Program Files\\\\$f\\\\cplex\\\\bin\\\\x86-64_win\\\\"
    elseif Sys.isapple()
        return "/Applications/$f/cplex/bin/x86-64_osx/"
    else
        return "/opt/$f/cplex/bin/x86-64_linux/"
    end
end
```

**Review by odow — COMMENTED** [review]
CI only works if you are on JuMP-dev and not a fork.

I can give you `Maintain` rights if you want? Here's what it entails https://jump.dev/pages/governance/#repository-maintainers

Then you'll be able to push to `jump-dev` and trigger CI.

**Comment by odow** [comment]
Any yes, it probably makes sense to try the default argument. No point recommending it if we can make it work.

File: deps/build.jl (hunk 1 above)
Comment (dourouc05) [reply deps/build.jl]: That's a remainder of a previous version of the code, I'm merging the three parts.

File: deps/build.jl (hunk 2 above)
Comment (dourouc05) [reply deps/build.jl:29]: Thanks, that's much nicer!

File: deps/build.jl [hunk 3]
```diff
@@ -16,6 +16,52 @@ function library_name(v)
     return "$(cpx_prefix)cplex$(v).$(Libdl.dlext)"
 end
 
+function get_error_message_if_not_found()
+    # Make the best guess for the CPLEX folder, based on the platform.
+    cplex_studio_folders = map(["CPLEX_Studio201", "CPLEX_Studio1210"]) do f
+        if Sys.iswindows()
+            return "C:\\\\Program Files\\\\IBM\\\\ILOG\\\\$f\\\\cplex\\\\bin\\\\x64_win64\\\\"
```
Comment (odow) [inline deps/build.jl:23]: Actually, I think you can go
```Julia
return escape_string("C:\\Program Files\\IBM\\ILOG\\$f\\cplex\\bin\\x64_win64\\")
```

**Comment by odow** [comment]
Did you want Maintain rights?

**Review by odow — APPROVED** [review]

**Comment by dourouc05** [comment]
The latest commit also uses the guessed path to check for CPLEX. It's only really tested for Windows, though.

I would be glad to obtain the maintainer rights, it will make things easier to contribute! By the way, maybe something similar can be achieved for CPLEXCP?

**Comment by odow** [comment]
I've sent you an invite.

cc @mlubin @blegat @joaquimg @joehuchette 

Setting up CI requires a secret environment variable that I'm not sure we can distribute outside of JuMP-dev due to license restrictions. What's the status of CPLEXCP? There's a [high bar for transferring repos to JuMP-dev](https://jump.dev/pages/governance/#transferring-repositories-to-jump-dev), but getting automated CI might meet that criteria.

**Comment by odow** [comment]
After you accept the invite, you should close this PR and open a new one on the `jump-dev` remote so we can see the CI run.

**Comment by dourouc05** [comment]
Currently, CPLEXCP is getting outside the state of prototype, with many MOI sets implemented (MILP, MISOCP, and parts of constraint programming). It's far from a 0.1 release, though. 

I'm closing this…

**Comment by odow** [comment]
Let us know when CPLEXCP is ready for a public release, and we can revisit the question of CI.

**Pull request closed without merging**